File: examples/05-production/python/worker_tuner.py
from temporalio.client import Client
from temporalio.worker import Worker, WorkerTuner


async def start(client: Client) -> None:
    # WorkerTuner.create_resource_based is the Python equivalent of Java's
    # ResourceBasedTuner: the worker hands out slots up to target CPU/memory
    # utilisation instead of a fixed count, so a busy host self-throttles.
    # target_* are fractions (0..1).
    tuner = WorkerTuner.create_resource_based(
        target_memory_usage=0.75,
        target_cpu_usage=0.80,
    )

    worker = Worker(
        client,
        task_queue="payments",
        workflows=[],
        activities=[],
        tuner=tuner,
    )
    await worker.run()

File: examples/05-production/go/worker_tuner.go
package production

import (
	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
)

// Worker tuning with a Tuner. NOTE on the difference vs. Java/Python: the Go SDK
// (v1.32) exposes fixed and composite tuners, but NOT the resource-based slot
// supplier that Java's ResourceBasedTuner / Python's WorkerTuner.create_resource_based
// provide. For CPU/memory-driven autosizing in Go you currently size slots
// explicitly (or use a CompositeTuner of fixed suppliers). Here we build a fixed
// tuner — the closest portable equivalent.
func StartTunedWorker(c client.Client) (worker.Worker, error) {
	tuner, err := worker.NewFixedSizeTuner(worker.FixedSizeTunerOptions{
		NumWorkflowSlots:      20,
		NumActivitySlots:      200,
		NumLocalActivitySlots: 20,
	})
	if err != nil {
		return nil, err
	}

	w := worker.New(c, "payments", worker.Options{Tuner: tuner})
	return w, nil
}
